### DISCUSSION TITLE
A modify to share ( on post )

### DISCUSSION THREAD

**Discussion by Vctrsnts** [sylhare/Type-on-Strap#442] [Q&A] [answered]
Hi.

I fork this beatiful theme and make one modify for make visible or hidden the share buttons on the post:

![Git_01](https://github.com/sylhare/Type-on-Strap/assets/59069334/3f3916cc-18be-4bb5-83e5-15a793e9cdf4)

The modify is in this two files:
- social.yml
- share_buttons.liquid

The code on `social.yml` is this:
```
# add this new option:
# For share on general
share_buttons: false
```
And the code on `share_buttons.liquid` is this:
```
# the firts line
{% if site.data.social.share_buttons %}
...
{% endif %}
```
Because I like the social ( mastodon, github, etc... ) but don't like the share option on the post and with this modify you have the social options, but without share.

Thanks a lot for this theme.

**Comment by sylhare** [comment]
I am not sure I understand, what would you like to do? Explícamelo en español si es más fácil.

I see that you already have some code, do you want to make a Pull request, that might be simpler to review.

**Reply by Vctrsnts** [reply]
De acuerdo.

Lo que digo, es que puedes tener configurado las redes sociales ( twitter, mastodon, etc... ) pero que en los post, tal como pongo en la imagen, no quieras que aparezca la opción de **SHARE** ( como se puede ver en la imagen tambien ).

Asi que lo que he hecho, ha sido añadir una pequeña modificación en los ficheros `social.yml` y `share_buttons.liquid`, para que, si se quiere, no visualizar el _SHARE y los iconos_.

En resumen, tienes configurado las redes sociales, pero la opción _SHARE_ que aparece en los post no se visualiza.

No se si ahora se me entiende mejor a lo que me estoy refiriendo.

De nuevo, gracias por todo.

**Reply by sylhare** [reply]
Ok gotcha, thanks for explaining!

Yes we could add a toggle to hide the "SHARE" text in front of the icon. 👍 

If you have it ready and want to make a PR I could integrate it to the theme sooner :) Else I'll add that when I find some time.

**Comment by Vctrsnts** [comment]
OK.

It is not to hide the SHARE text, but rather it is to hide the entire SHARE section (icons included).

Regarding the PR, I don't feel very comfortable with them, I'm afraid 😂  that I will break some of the code by doing so, so here is the necessary code available for when you have time.

And above all, thanks to you for this incredible work.

**Reply by sylhare** [reply, marked as answer]
Oh If you don't want any of the icon you can do in `_data/social`.
This will remove the "share" button and the icons.

```
# For the share button options at the bottom of the articles
share:
  email: false
  facebook: false
  linkedin: false
  pinterest: false
  pocket:
  reddit: false
  tumblr: false
  twitter: false
  wordpress:
```  

It's different from the `icons` that are at the bottom of the page like `mastodon`, `github`, ...

Give it a try and let me know if that matches what you want to do 👍

**Comment by Vctrsnts** [comment]
You're right.

I have carried out tests with what you have told me and if it is as you say.

If you set everything to `false` in the `social.yml` file, no **SHARE** icon appears at the end of the articles.

Again, sorry, my fault for not paying more attention to the code.

So, thanks to you for all the help you have given me and for correcting my mistake.

Thank you

**Reply by sylhare** [reply]
Perfect, then I'll close the discussion 😄
